File: Problem_003/C++/main.cpp
// 3. Longest Substring Without Repeating Characters 
// Given a string s, find the length of the longest substring without repeating characters.

#include <iostream>
#include <string>
#include <unordered_map>
#include "Solution.h"

int main() {
	
	Solution s;
	std::string testString = "abcabcbb";
	
	int answer = s.lengthOfLongestSubstring(testString);	

	std::cout << "main answer: " << answer << std::endl;
	
	return 0;

}

File: Problem_003/Python/main.py
# 3. Longest Substring Without Repeating Characters
# Given a string s, find the length of the longest substring without repeating characters.  

from Solution import Solution

def main():
	
	testString = "abcabcbb"

	# instantiate Solution class
	s = Solution()

	# Solution returns an int length of the longest non-repeating substring in the testString
	answer = s.lengthOfLongestSubstring(testString)

if __name__ == "__main__":
	main()
